packages/block-editor/src/store/selectors.js

```diff
@@ -139,9 +147,15 @@ export const getBlock = createSelector(
 		return {
 			...block,
 			attributes: getBlockAttributes( state, clientId ),
-			innerBlocks: areInnerBlocksControlled( state, clientId )
-				? EMPTY_ARRAY
-				: getBlocks( state, clientId ),
+			innerBlocks:
+				! includeControlledInnerBlocks &&
+				areInnerBlocksControlled( state, clientId )
+					? EMPTY_ARRAY
```

Comment:
This is a bug I think, since this won't apply if there's a block with innerBlocks controlled deep in the tree right?

Reply:
I think it should still work since we would be recursing down and continuing to pass `includeControlledInnerBlocks`?

I think this logic says:

> If we are in "exclude" controlled inner blocks mode, and if the current block is an inner block controller, then do not return the inner blocks.

That same logic should still apply if this is deep in the tree, because the argument is passed down through the recursing calls.

does that make sense?

Reply:
Seems like  you're right 👍